libs/common/grpc.go

```diff
@@ -146,16 +146,24 @@ func authFunc(ctx context.Context) (context.Context, error) {
 // organizationID from the headers against the organizationIDs inside the claim.
 func handleOrganizationIDForAuthFunc(ctx context.Context) (context.Context, error) {
 	organizationIDStr, err := OrganizationIDFromMD(ctx)
-	if err != nil {
+
+	var organizationID uuid.UUID
+	if err == nil {
+		// organizationID in header, using header
+		organizationID, err = uuid.Parse(organizationIDStr)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "invalid organizationID")
+		}
+	} else if OrganizationID != nil {
+		// no organizationID in header but in envs, using env
+		zlog.Ctx(ctx).Info().Msg("no valid organization header found, falling back to environment organization id")
```

Comment:
```suggestion
		zlog.Ctx(ctx).Trace().Msg("no valid organization header found, falling back to environment organization id")
```

Reply:
also there are now two instances where a logger is used, so we should declare a log variable at the top of the function instead